File: app/src/main/java/com/example/dynamodbapp2/Myadapter.java
package com.example.dynamodbapp2;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

public class Myadapter extends BaseAdapter {

    Context context;

    public Myadapter(Context context){
        this.context = context;
    }

    @Override
    public int getCount() {
        return GenerateKeys.keysholder.size();
    }

    @Override
    public Object getItem(int position) {
        return position;
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View row = convertView;
        viewHolder viewHolder = null;
        if (row==null){
            LayoutInflater layoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            row = layoutInflater.inflate(R.layout.singlerow, parent, false);
            viewHolder = new viewHolder(row);
            row.setTag(viewHolder);
        }else {
            viewHolder = (Myadapter.viewHolder) row.getTag();
        }

        viewHolder.textView.setText(GenerateKeys.keysholder.get(position).get("key").toString());
        return null;
    }

    private class viewHolder{
        TextView textView;
        public viewHolder(View view){
            textView = view.findViewById(R.id.singletextview);
        }
    }
}

File: app/src/main/java/com/example/dynamodbapp/MyAdapter.kt
package com.example.dynamodbapp

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import kotlinx.android.synthetic.main.singlerow.view.*

class MyAdapter(context: Context) : BaseAdapter() {

    val context = context

    override fun getCount(): Int {
        return GenerateKeys.keys_holder.size
    }

    override fun getItem(position: Int): Any {
        return position
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View? {
        var row = convertView
        var viewHolder : ViewHolder? = null
        if (row==null){
            val layoutInflater: LayoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            row = layoutInflater.inflate(R.layout.singlerow, parent, false)
            viewHolder = ViewHolder(row)
            row.tag = viewHolder
        }
        else{
            viewHolder = row.getTag() as ViewHolder
        }

        viewHolder.textView.setText(GenerateKeys.keys_holder.get(position).get("keys").toString())
        return null
    }

    private class ViewHolder(view : View){
        var textView: TextView
        init {
            textView = view.singletextview
        }
    }
}